lib/shared-method.js

```diff
@@ -236,24 +236,25 @@ function escapeRegex(d) {
  * @return {*}                 Coerced argument.
  */
 function coerceAccepts(uarg, desc) {
-  // If array, invoke for each member of the array.
-  if (Array.isArray(uarg)) {
-    return uarg.map(function(arg) {
-      return coerceAccepts(arg, desc);
-    });
-  }
-
   var name = desc.name || desc.arg;
   var targetType = desc.type;
-  var actualType = SharedMethod.getType(uarg);
-
-  // Target type may be an array. If so, expect the arg to be a member of that array,
-  // since we map this function over arrays.
   var targetIsArray = Array.isArray(targetType) && targetType.length === 1;
-  if (targetIsArray) {
-    targetType = targetType[0];
+
+  // If coercing an array to an erray,
+  // then coerce all members of the array too
+  if (targetIsArray && Array.isArray(uarg)) {
+    return uarg.map(function(arg, ix) {
+      // when coercing array items, use only name and type,
+      // ignore all other root settings like "required"
+      return coerceAccepts(arg, {
+        name: name + '[' + ix +']',
+        type: desc.type[0]
```

Comment:
I agree that such object would be great. However, I don't have time to make this refactoring now :(